lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp

```diff
@@ -1124,138 +1124,38 @@ void mlir::torch::onnx_c::populateDefaultDomainGtoP(
       });
   patterns.onOp(
       "MaxPool", 12, [](OpBinder binder, ConversionPatternRewriter &rewriter) {
-        std::string autoPad;
-        if (binder.customOpNameStringAttr(autoPad, "auto_pad", "NOTSET"))
-          return rewriter.notifyMatchFailure(binder.op,
-                                             "auto_pad bind failure");
-
         Torch::ValueTensorType resultTypeOut;
         Value operand;
         int64_t ceilMode, storageOrder;
-        // TODO: Add support for indices output and storage_order
+        std::string autoPad;
         if (binder.tensorOperand(operand) ||
             binder.s64IntegerAttr(ceilMode, "ceil_mode", 0) ||
             binder.s64IntegerAttr(storageOrder, "storage_order", 0) ||
+            binder.customOpNameStringAttr(autoPad, "auto_pad", "NOTSET") ||
             binder.tensorResultTypeAtIndex(resultTypeOut, 0))
           return rewriter.notifyMatchFailure(
-              binder.op,
-              "operand/ceil_mode/storage_order/resultType bind failure");
+              binder.op, "operand/ceil_mode/storage_order/auto_pad/resultType "
+                         "bind failure");
+        // TODO: Add support for storage_order
         if (storageOrder != 0)
           return rewriter.notifyMatchFailure(
               binder.op, "storage_order setting is not supported.");
+
         // Determine the rank of input tensor.
         std::optional<unsigned> maybeRank = Torch::getTensorRank(operand);
         if (!maybeRank)
           return rewriter.notifyMatchFailure(binder.op,
                                              "Unimplemented: unranked tensor");
-        int64_t rank = *maybeRank;
-        int64_t spatial = rank - 2;
+        unsigned rank = *maybeRank;
 
-        SmallVector<int64_t> kernel, padding, strides, dilations;
-        if (binder.s64IntegerArrayAttr(kernel, "kernel_shape", {}))
+        SmallVector<int64_t> kernel, padding, strides, dilations,
+            stridesDilations;
+        if (failed(checkAndGetOnnxPoolingOpParameters(
```

Comment:
How about `countIncludePad = false`? If we pad using `AtenConstantPadNdOp `and after that do `AtenAvgPoolOp `don’t we lose the ability to support `countIncludePad = false`?